File: didatico/julia/codes/min1.f90
program min1
  implicit none
  double precision :: x, dfdx, deltax, deltaf, xbest, fbest
  deltax = 0.1d0 ! Step size
  x = 14.1357d0 ! Initial guess
  fbest = x**2
  xbest = x ! Save best point
  write(*,*) ' Initial point: '
  write(*,*) x, x**2
  do
    ! Move x in the descent direction, with step deltax
    dfdx = 2.d0*x ! Computing the derivative
    x = x - deltax * dfdx/dabs(dfdx) ! Move x in the -f' direction
    ! Test new point
    deltaf = x**2 - fbest
    ! Write current point
    write(*,*) x, x**2, deltaf
    ! If the function decreased, save best point
    if ( deltaf < 0 ) then
      xbest = x
      fbest = x**2
    else
      write(*,*) ' Function is increasing. '
      write(*,*) ' Best solution found: x = ', xbest
      exit
    end if
  end do
end program min1

File: main/didatico/simulacoes/julia/min1.jl
function min1(x0)
  x = x0
  deltax = 0.1 # Step size
  xbest = x # Save best point
  fbest = x^2 # Best value up to now
  println(" Initial point: ",xbest," ",fbest)
  deltaf = -1.
  while deltaf < 0. 
    # Move x in the descent direction, with step deltax
    dfdx = 2*x # Computing the derivative
    x = x - deltax * dfdx/abs(dfdx) # Move x in the -f' direction
    # Test new point
    deltaf = x^2 - fbest
    # Write current point
    println("Current point: ",x," ",x^2," ",deltaf)
    # If the function decreased, save best point
    if deltaf < 0.
      xbest = x
      fbest = x^2
    else
      println(" Function is increasing. ")
      println(" Best solution found: x = ", xbest, " f(x) = ",fbest)
      return xbest, fbest
    end 
  end
end
